test/timer_test_grain.erl

```diff
@@ -0,0 +1,125 @@
+%%% ---------------------------------------------------------------------------
+%%% %%% @copyright 2017 Space-Time Insight <tristan.sloughter@spacetimeinsight.com>
+%%%
+%%% @doc A test grain that sets up some timers and then accumulates
+%%% the results of messages sent to it.
+%%% @end
+%%% ---------------------------------------------------------------------------
+-module(timer_test_grain).
+
+-behaviour(erleans_grain).
+
+-export([placement/0,
+         provider/0,
+         save/1,
+         node/1,
+         stop/1,
+         start_timers/1,
+         cancel_timers/1,
+         start_one_timer/1,
+         cancel_one_timer/1,
+         accumulate/2,
+         clear/1]).
+
+-export([init/2,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         deactivate/1]).
+
+-include("erleans.hrl").
+
+placement() ->
+    prefer_local.
+
+provider() ->
+    erleans_config:get(default_provider).
+
+start_timers(Ref) ->
+    erleans_grain:call(Ref, start_timers).
+
+cancel_timers(Ref) ->
+    erleans_grain:call(Ref, cancel_timers).
+
+start_one_timer(Ref) ->
+    erleans_grain:call(Ref, start_one_timer).
+
+cancel_one_timer(Ref) ->
+    erleans_grain:call(Ref, cancel_one_timer).
+
+accumulate(Ref, Thing) ->
+    erleans_grain:call(Ref, {accumulate, Thing}).
+
+clear(Ref) ->
+    erleans_grain:call(Ref, clear).
+
+save(Ref) ->
+    erleans_grain:call(Ref, save).
+
+node(Ref) ->
+    erleans_grain:call(Ref, node).
+
+stop(Ref) ->
+    erleans_grain:call(Ref, stop).
+
+init(_, State=#{}) ->
+    {ok, State#{acc => []}, #{life_time => infinity}}.
+
+handle_call(node, _From, State) ->
+    {reply, {ok, node()}, State};
+
+handle_call(clear, _From, State = #{acc := Acc}) ->
+    {reply, {ok, Acc}, State#{acc => []}};
+handle_call({accumulate, Thing}, _From, State=#{acc := Acc}) ->
+    {reply, ok, State#{acc => [Thing | Acc]}};
+
+handle_call(start_timers, _From, State) ->
+    Ref = get(grain_ref),
+    F1 = fun() ->
```

Comment:
OK.  The API there is a little tricky.  Do we make it explicit? That means that a grain could potentially create timers for another grain.  Do we want to allow that?  Otherwise we could implicitly pick it up from the pdict, but that would mean that it would only be possible to create a time from grain context, which might be something that we're OK with (I am, I lean towards this option).